Formula/cvs-fast-export.rb

```diff
@@ -14,6 +20,7 @@ class CvsFastExport < Formula
   end
 
   depends_on "asciidoc" => :build
+  depends_on "bison" => :build
```

Comment:
```suggestion
  depends_on "bison" => :build if build.head?
```
Since you only need this for the head build?

Reply:
Good suggestion, but reading [this issue](https://gitlab.com/esr/cvs-fast-export/-/issues/18) from `cvs-fast-export`, it's clear that the upstream expects `bison` 3 or newer; so I think it's safer (ie more future-proof) to keep this in the base build as well.

When I originally wrote the formula, I had problems with the build not working reliably without `docbook-xsl` - I now believe I missed this build dependency when I first submitted.

OK?

Reply:
`depends_on "bison" => :build if build.head?` didn't work when I tested it. `build.head?` works in the `install` block but doesn't seem to work in this outer scope, so `bison` doesn't get included when `if build.head?` is added to this statement.

The way I usually handle this is to add the `depends_on` call in a `head` block, like:

```ruby
head do
  url "https://gitlab.com/esr/cvs-fast-export.git"
  depends_on "bison" => :build
end
```

This worked as expected when I tested it.

The trouble is that the `head` block comes after the `livecheck` block in the component order, so using `url :head` in the `livecheck` block with this setup won't work. I'll be creating a Homebrew/brew PR in a moment to modify the component order, though there's no guarantee it will be merged. [If it does get merged, we'll need to update formulae with `head` blocks to move them into the new position.]

Reply:
As I noted, `depends_on "bison" => :build` really should be part of the base formula, since the upstream package assumes `bison` version 3... is this PR acceptable as it currently stands?

Reply:
Sorry about that, I hadn't refreshed the page before commenting so I missed your previous comment.

From what I've seen, building from source using the `stable` tarball (1.55) looks exactly the same with or without `depends_on "bison" => :build`. `bison` is necessary when doing a `HEAD` build but it doesn't seem to be required when using the latest release archive.

The notable difference between the tarball from the first-party website and the files on GitLab is that GitLab is missing `gram.c`, `gram.h`, `lex.c`, and `lex.h`. Looking at the `Makefile`, Bison is involved with the `gram` files, so that makes sense.

If Bison will be required to build the next version after `1.55` (using a release tarball like we're currently doing), then it would make sense to add `depends_on "bison" => :build` without restricting it to `HEAD` builds. However, I'm not sure how we generally feel about preemptively adding dependencies before they're explicitly needed. I imagine we probably lean toward, "Wait to add the dependency until it's actually needed" (i.e., add the dependency as part of the next version bump if it's needed at that time).

That said, I'm not familiar with this formula/software, so please correct my understanding if I've misunderstood.

Reply:
OK, given all that it does make sense to restrict the `depends_on` to just `HEAD` for now. Should I wait for https://github.com/Homebrew/brew/pull/10157 or just implement the requested changes?

Reply:
My suggested `livecheck` block above doesn't use the Git repository, so we aren't technically blocked by the brew PR. I can merge this if we do the following:

* Integrate the suggested `livecheck` block (as it aligns the check with the source of the `stable` archive and we do this whenever possible).
* Remove `head` and add the following `head` block below the `bottle` block:

```ruby
head do
  url "https://gitlab.com/esr/cvs-fast-export.git"
  depends_on "bison" => :build
end
```

Reply:
Thanks for the clarification! I've resolved both of these, can you review?